subsys/net/lib/nrf_cloud/Kconfig

```diff
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-if NRF_CLOUD_MQTT || NRF_CLOUD_REST || NRF_CLOUD_PGPS
+if NRF_CLOUD_MQTT || NRF_CLOUD_REST || NRF_CLOUD_AGPS || NRF_CLOUD_PGPS
```

Comment:
This change is needed because NRF_CLOUD_SEC_TAG is used in nrf_cloud_jwt_generate, which is compiled with AGPS even if REST and MQTT are turned off (we have the use case of external AGPS source).

Reply:
`nrf_cloud_jwt_generate()` is compiled when `CONFIG_MODEM_JWT` is defined. Is that needed with an external A-GPS source, probably not? Just thinking whether adding the condition for `CONFIG_NRF_CLOUD_AGPS` is the right thing to do, because that flag is not directly related to `nrf_cloud_jwt_generate()` compilation.

Reply:
I agree that CONFIG_MODEM_JWT is not needed with an external A-GPS source. That being said, I suppose it may well be enabled.

Reply:
Yes, but should the used flag the be `MODEM_JWT` instead of `NRF_CLOUD_AGPS`?

Reply:
I actually didn't pay attention to `CONFIG_MODEM_JWT` flagging in this file. I just looked at `CMakeFiles.txt` which compiles `nrf_cloud_jwt.c` always but it becomes empty based on `CONFIG_MODEM_JWT`. I'll actually modify `CMakeFiles.txt` and `nrf_cloud_jwt.c` also.

> Just thinking whether adding the condition for `CONFIG_NRF_CLOUD_AGPS` is the right thing to do, because that flag is not directly related to `nrf_cloud_jwt_generate()` compilation.

I'll change this to `CONFIG_MODEM_JWT`. However, I'm not sure other flags used here are correct either but not sure what the condition should be given there is host name in addition to sec tag.